vm/src/builtins/union.rs

```diff
@@ -156,6 +162,8 @@ fn flatten_args(args: PyTupleRef, vm: &VirtualMachine) -> PyTupleRef {
     for arg in &args {
         if let Some(pyref) = arg.downcast_ref::<PyUnion>() {
             flattened_args.extend(pyref.args.iter().cloned());
+        } else if vm.is_none(arg) {
+            flattened_args.push(vm.ctx.types.none_type.as_ref().to_pyobject(vm).clone());
```

Comment:
```suggestion
            flattened_args.push(vm.ctx.types.none_type.to_owned().into());
```
this is the last piece